library-best-practices/JLBP-6.md

```diff
@@ -0,0 +1,141 @@
+[JLBP-6] Avoid creating conflicts between major versions
+------------------------------------------------------------------------
+
+- A new major version should generally use a different Java package.
+- There is a tradeoff: if an extremely large library (for example thousands of
+  classes) breaks a single method, the cost of a package rename may be higher
+  than dealing with the fallout of dealing with diamond dependency conflicts for
+  that particular method. Only perform an in-place breakage if users can easily
+  replace their usage of the old library with the new library. The larger the
+  portion of a library that breaks between major versions, the higher the value
+  is from renaming the java package.
+- *If* the Java package is changed, use different maven coordinates (for example
+  different artifact id).
+- *If* the Java package is kept the same, use the same maven artifact ID.
+- In other words, rename the Java package if and only if the maven coordinates
+  change.
+
+Background
+----------
+
+When a library B in Java depends on another library A using Apache Maven, there
```

Comment:
seems to be about using the maven repo system, not necessarily maven

Reply:
Good point, will fix.